### PR TITLE
Implement sagas for save and upsate of SUMA settings

### PR DESCRIPTION
**Pull request by dottorblaster**
# Description
As above, just implementing the sagas to save and update SUMA settings :+1: 

## How was this tested?
Jest tests

### CODE REVIEW

File: assets/js/state/sagas/softwareUpdatesSettings.test.js [hunk 1]
```diff
@@ -44,4 +51,134 @@ describe('Software Updates Settings saga', () => {
...
+
+    it('should have errors on failed saving', async () => {
+      const axiosMock = new MockAdapter(networkClient);
+      const payload = {
+        url: '',
+        username: '',
+        password: faker.word.noun(),
+        ca_cert: '',
+      };
+      const errors = [
```
Comment (nelsonkopliku) [inline assets/js/state/sagas/softwareUpdatesSettings.test.js:91]: OCD alert: mismatching errors based on the input. 
Providing an empty username `username: '',` would result in an extra error besides url/ca_cert.

Feel free to keep like this, tho :smile:  The test still tests what it's supposed to test.

File: assets/js/state/sagas/softwareUpdatesSettings.test.js [hunk 2]
```diff
@@ -44,4 +51,134 @@ describe('Software Updates Settings saga', () => {
...
+
+      expect(dispatched).toEqual([
+        startLoadingSoftwareUpdatesSettings(),
+        setSoftwareUpdatesSettingsErrors(errors),
+      ]);
+    });
+  });
+
+  describe('Updating Software Updates settings', () => {
+    it('should successfully save software updates settings', async () => {
```
Comment (nelsonkopliku) [inline assets/js/state/sagas/softwareUpdatesSettings.test.js:118]: maybe a copy/pasta from previous test
```suggestion
    it('should successfully change software updates settings', async () => {
```

File: assets/js/state/sagas/softwareUpdatesSettings.test.js [hunk 3]
```diff
@@ -44,4 +51,134 @@ describe('Software Updates Settings saga', () => {
...
+        payload
+      );
+
+      expect(dispatched).toEqual([
+        startLoadingSoftwareUpdatesSettings(),
+        setSoftwareUpdatesSettings(successfulResponse),
+      ]);
+    });
+
+    it('should have errors on failed saving', async () => {
```
Comment (nelsonkopliku) [inline assets/js/state/sagas/softwareUpdatesSettings.test.js:148]:
```suggestion
    it('should have errors on failed update', async () => {
```

**Review by nelsonkopliku — APPROVED** [review]
LGTM.
Only test names might be specified a bit, but not a blocker :+1:

**Review by jamie-suse — APPROVED** [review]

**Pull request merged**